### PR TITLE
`struct RefMvsFrame::r`: Add `DisjointMut`

### PR DESCRIPTION
**Pull request by kkysen**
This removes `&mut` accesses of `Rav1dFrameData` added in #942.

### CODE REVIEW

File: src/refmvs.rs [hunk 1]
```diff
@@ -362,15 +363,35 @@ pub(crate) struct Rav1dRefmvsDSPContext {
...
+            }
+            let ri = rt.r[i];
+            if ri > rf.r.len() {
+                return Default::default();
+            }
+            // This is the range that will actually be accessed,
+            // but `splat_mv` expects a pointer offset `bx4` backwards.
+            rf.r.index_mut(ri + bx4..ri + bx4 + bw4)
+        });
+        let mut r: [_; 37] = array::from_fn(|i| {
```
Comment (rinon) [inline src/refmvs.rs:387]: This drops the guard by re-using the name, right? Shouldn't we hold the guard across the call to `splat_mv`?

File: src/refmvs.rs [hunk 2]
```diff
@@ -362,15 +363,35 @@ pub(crate) struct Rav1dRefmvsDSPContext {
...
+            if r[i].is_empty() {
```
Comment (rinon) [inline src/refmvs.rs:388]: Couldn't we have used `Option<DisjointMutGuard<...>>` instead?

**Review by rinon — COMMENTED** [review]
Was it just the one place that needed a default impl for the guards? It seems weird to impl default unless we absolutely need to. Option should be basically the same?

File: src/refmvs.rs (hunk 1 above)
Comment (kkysen) [reply src/refmvs.rs:387]: No, `array::from_fn` creates a new array without consuming the old one, unlike `.map`.

**Comment by kkysen** [comment]
> Was it just the one place that needed a default impl for the guards? It seems weird to impl default unless we absolutely need to. Option should be basically the same?

I think so.  I was previously using it somewhere else, too, where `impl Default`ing it was more useful so we could avoid `.unwrap()`s in release code, but that had to be removed/reworked for other reasons, so I think I could use an `Option` as you suggested, which is definitely a lot simpler.

File: src/refmvs.rs (hunk 2 above)
Comment (kkysen) [reply src/refmvs.rs]: Yup, switched to `Option` now.

**Review by randomPoison — APPROVED** [review]

File: src/refmvs.rs [hunk 3]
```diff
@@ -464,15 +483,17 @@ fn scan_row(
     cnt: &mut usize,
     r#ref: refmvs_refpair,
     gmv: &[mv; 2],
-    b: &[refmvs_block],
+    r: &DisjointMut<AlignedVec64<refmvs_block>>,
+    b_offset: usize,
+    // b: &[refmvs_block],
```
[the remark this answers is not in the document]
Comment (kkysen) [reply src/refmvs.rs:488]: Nope, just forgot to remove it.

**Pull request merged**